cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowStatsSLAHttpHandler.java

```diff
@@ -89,4 +108,131 @@ public void workflowStats(HttpRequest request, HttpResponder responder,
     }
     responder.sendJson(HttpResponseStatus.OK, workflowStatistics);
   }
+
+  @GET
+  @Path("apps/{app-id}/workflows/{workflow-id}/runs/{run-id}/statistics")
+  public void workflowRunDetail(HttpRequest request, HttpResponder responder,
+                                @PathParam("namespace-id") String namespaceId,
+                                @PathParam("app-id") String appId,
+                                @PathParam("workflow-id") String workflowId,
+                                @PathParam("run-id") String runId,
+                                @QueryParam("count") int count,
+                                @QueryParam("interval") String interval) throws Exception {
+    long timeInterval = TimeMathParser.resolutionInSeconds(interval);
+    Id.Workflow workflow = Id.Workflow.from(Id.Namespace.from(namespaceId), appId, workflowId);
+    Collection<WorkflowDataset.WorkflowRunRecord> workflowRunRecords =
+      store.retrieveSpacedRecords(workflow, runId, count, timeInterval).values();
+    Map<String, WorkflowStatistics.DetailedStatistics> detailedStatisticsMap = new HashMap<>();
+    for (WorkflowDataset.WorkflowRunRecord workflowRunRecord : workflowRunRecords) {
+      WorkflowStatistics.DetailedStatistics detailedStatistics =
+        getDetailedRecord(workflow, workflowRunRecord.getWorkflowRunId());
+      detailedStatisticsMap.put(workflowRunRecord.getWorkflowRunId(), detailedStatistics);
+    }
+    Map<String, Map<String, Map<String, Long>>> formattedStatisticsMap = convert(detailedStatisticsMap);
+
+    responder.sendJson(HttpResponseStatus.OK, formattedStatisticsMap);
+  }
+
+  @GET
+  @Path("apps/{app-id}/workflows/{workflow-id}/compare/run-id/{run-id}")
+  public void compare(HttpRequest request, HttpResponder responder,
+                      @PathParam("namespace-id") String namespaceId,
+                      @PathParam("app-id") String appId,
+                      @PathParam("workflow-id") String workflowId,
+                      @PathParam("run-id") String runId,
+                      @QueryParam("other-run-id") String otherRunId) throws Exception {
+    Id.Workflow id = Id.Workflow.from(Id.Namespace.from(namespaceId), appId, workflowId);
+    WorkflowStatistics.DetailedStatistics detailedStatistics = getDetailedRecord(id, runId);
+    WorkflowStatistics.DetailedStatistics otherDetailedStatistics = getDetailedRecord(id, otherRunId);
+    if (detailedStatistics == null) {
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, "The run-id you provided was not correct.");
```

Comment:
Better to not use a notion of 'person' in error messages. Also, why is this not correct? Because it was not found? Then let's return ``HttpResponseStatus.NOT_FOUND``. Also include the runid in the error message